### ISSUE TITLE
mvn:build stuck on GraalVM 21.0.0

### ISSUE THREAD

**Issue by makingthematrix** [gluonhq/client-maven-plugin#289]
Hi,

I've just updated GraalVM to the new version and tried to build an example application with it. This is how far it got before hanging:
```
[INFO] --- client-maven-plugin:0.1.35:build (default-cli) @ scalaonandroid ---
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$1 (file:/usr/share/maven/lib/guice.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[INFO] Scanning for projects...
[INFO] 
[INFO] -----------------< io.makingthematrix:scalaonandroid >------------------
[INFO] Building Scala 2.13 on Android 0.0.2
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] >>> client-maven-plugin:0.1.35:compile (default-cli) > process-classes @ scalaonandroid >>>
[INFO] 
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ scalaonandroid ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 5 resources
[INFO] 
[INFO] --- scala-maven-plugin:4.4.0:add-source (scala-compile) @ scalaonandroid ---
[INFO] Add Source directory: /home/makingthematrix/workspace/scalaonandroid/src/main/scala
[INFO] Add Test Source directory: /home/makingthematrix/workspace/scalaonandroid/src/test/scala
[INFO] 
[INFO] --- scala-maven-plugin:4.4.0:compile (scala-compile) @ scalaonandroid ---
[INFO] Using incremental compilation using Mixed compile order
[INFO] Compiler bridge file: /home/makingthematrix/.sbt/1.0/zinc/org.scala-sbt/org.scala-sbt-compiler-bridge_2.13-1.3.5-bin_2.13.4__55.0-1.3.5_20200330T000931.jar
[INFO] Compiling 8 Scala sources to /home/makingthematrix/workspace/scalaonandroid/target/classes ...
[WARNING] : 1 deprecation (since 2.13.0); re-run with -deprecation for details
[WARNING] one warning found
[INFO] Done compiling.
[INFO] compile in 6.6 s
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ scalaonandroid ---
[INFO] Changes detected - recompiling the module!
[INFO] 
[INFO] <<< client-maven-plugin:0.1.35:compile (default-cli) < process-classes @ scalaonandroid <<<
[INFO] 
[INFO] 
[INFO] --- client-maven-plugin:0.1.35:compile (default-cli) @ scalaonandroid ---
```

Don't mind the Scala part. When I switched back to GraalVM 20.0.3, it went smoothly through that step where it hangs with GraalVM 21:
```
[INFO] --- client-maven-plugin:0.1.35:compile (default-cli) @ scalaonandroid ---
[Wed Jan 20 12:39:04 CET 2021][INFO] ==================== COMPILE TASK ====================
             _______  ___      __   __  _______  __    _
            |       ||   |    |  | |  ||       ||  |  | |
            |    ___||   |    |  | |  ||   _   ||   |_| |
            |   | __ |   |    |  |_|  ||  | |  ||       |
            |   ||  ||   |___ |       ||  |_|  ||  _    |
            |   |_| ||       ||       ||       || | |   |
            |_______||_______||_______||_______||_|  |__|

    Access to the latest docs, tips and tricks and more info on
    how to get support? Register your usage of Gluon Substrate now at

    https://gluonhq.com/activate



[Wed Jan 20 12:39:05 CET 2021][INFO] We will now compile your code for x86_64-linux-linux. This may take some time.
[Wed Jan 20 12:39:10 CET 2021][INFO] [SUB] [io.makingthematrix.scalaonandroid.europeanunion.main:4753]    classlist:   2,577.56 ms,  0.96 GB
[Wed Jan 20 12:39:11 CET 2021][INFO] [SUB] [io.makingthematrix.scalaonandroid.europeanunion.main:4753]        (cap):     661.18 ms,  0.96 GB
[Wed Jan 20 12:39:12 CET 2021][INFO] [SUB] [io.makingthematrix.scalaonandroid.europeanunion.main:4753]        setup:   2,216.96 ms,  0.96 GB
```
... and so on. Compilation proceeds as it should.

For a moment I thought the problem might be about that warning at the beginning of the compilation log, "an illegal reflective access", but it seems to be another issue: https://github.com/google/guice/issues/1133

I'm under Linux. I use GraalVM for Java 11 and AMD64 architecture. Here's the list of properties from my pom.xml:
```
    <properties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <maven.compiler.release>11</maven.compiler.release>
        <javafx.version>15.0.1</javafx.version>
        <javafx.plugin.version>0.0.5</javafx.plugin.version>
        <client.plugin.version>0.1.35</client.plugin.version>
        <charm.version>6.0.6</charm.version>
        <attach.version>4.0.10</attach.version>
        <main.class>io.makingthematrix.scalaonandroid.europeanunion.Main</main.class>
        <scala.version>2.13.4</scala.version>
        <scala.compat.version>2.13</scala.compat.version>
    </properties>
```

If you want, you can download the example app I try to build from here: https://github.com/makingthematrix/scalaonandroid

Thanks!

PS. Or is it something I should report to GraalVM?

**Comment by jperedadnr** [comment]
We have just released 0.1.36 btw, which requires GraalVM 21.0.0.

Does it work for you if you run a HelloXX sample from https://github.com/gluonhq/gluon-samples ?

When you say "hangs", do you mean it doesn't even print
```
[Wed Jan 20 12:39:04 CET 2021][INFO] ==================== COMPILE TASK ====================
``` 
?

Can you try with `<verbose>true</verbose>` in the plugin's configuration?

**Comment by makingthematrix** [comment]
@jperedadnr : That's weird. I was able to reproduce this problem 100% yesterday, but today it works, even before updating the plugin. I made a few tests: 0.1.36 works with GraalVM 21 for both my proejct and the HelloFX sample, but also 0.1.35 works with both GraalVM 20.3.0 and 21 on both. Now I think maybe it was a Maven problem, maybe it tried to contact the server for something that was missing.

Yes, that line "=== COMPILE TASK ===" wasn't printed at all. But it's ok now. 

Sorry for wasting your time. I'll close this issue now :)
